node/src/actors/chain_manager/mod.rs

```diff
@@ -542,26 +542,17 @@ impl ReputationInfo {
     ) {
         let dr_pointer = tally_transaction.dr_pointer;
         let dr_state = &data_request_pool.data_request_pool[&dr_pointer];
-        let reveals = &dr_state.info.reveals;
         let commits = &dr_state.info.commits;
         let replication_factor = dr_state.data_request.witnesses;
         self.alpha_diff += Alpha(u32::from(replication_factor));
 
-        let _tally = &tally_transaction.tally;
-
+        let mut tally_vts = HashSet::new();
+        // TODO: Review case of data requester is also a witness
```

Comment:
With the current implementation, the data requester can lie with no penalization if the tally transaction contains a change output. It would be nice to have a test for this case.

Reply:
Yes, it happens. But you say a test for check that this bug happens or doesnt happen?

Reply:
Ideally I think we should fix that so it does not happen. But I don't know how straightforward would that be.

Reply:
Created #903